client/utils/currency/index.js

```diff
@@ -80,15 +103,33 @@ export const formatCurrency = ( amount, currencyCode = 'USD' ) => {
 
 	const currency = getCurrency( currencyCode );
 	if ( null === currency ) {
-		return composeFallbackCurrency( amount, currencyCode, isZeroDecimal );
+		return composeFallbackCurrency(
+			amount,
+			omitCode ? null : currencyCode,
+			isZeroDecimal
+		);
 	}
 
 	try {
-		return 'function' === typeof currency.formatAmount
-			? currency.formatAmount( amount )
-			: currency.formatCurrency( amount );
+		const formattedAmount =
+			'function' === typeof currency.formatAmount
+				? currency.formatAmount( amount )
+				: currency.formatCurrency( amount );
+
+		if (
+			omitCode ||
+			wcSettings.currency.code === currencyCode.toUpperCase()
+		) {
+			return formattedAmount;
+		}
+
+		return `${ currencyCode.toUpperCase() } ${ formattedAmount }`;
```

Comment:
Is there a way for us to respect the [`currency.symbolPosition`](https://github.com/woocommerce/woocommerce-admin/blob/e9f0f8daf3a5989090054f1bd6c209bc365cbbcc/packages/currency/src/index.js#L229) here? Not all currency codes should appear before the amount.

That said, maybe we should respect the store's settings here? I think it might be better to think of what the merchant will think is least confusing — if they're used to the currency symbol being in front of the amount, it might be better to respect that setting, rather than where the currency symbol is canonically placed.

On the other hand, if a merchant is familiar with other conventions (such as DKK/NOK/SEK/ISK currencies being written `<amount> kr.` or `<amount> <currency_code>`) that might also throw them off.

What do you think @LevinMedia ?